tests/check_deprecations.py

```diff
@@ -0,0 +1,178 @@
+    if len(values) != 3:
+        raise ValueError(f"Unexpected number of version elements ({len(values)}): {values}")
+    return tuple(values)
+
+
+def get_deprecated_methods_that_should_have_been_removed(codebase_path: str,
+                                                         decorator_name: str,
+                                                         medcat_version: Tuple[int, int, int]
+                                                         ) -> List[Tuple[str, str, Tuple[int, int, int]]]:
+    """Get deprecated methods that should have been removed.
+
+    Args:
+        codebase_path (str): Path to codebase.
+        decorator_name (str): Name of decorator to check.
+        medcat_version (Tuple[int, int, int]): The current MedCAT version.
+
+    Returns:
+        List[Tuple[str, str, Tuple[int, int, int]]]:
+            The list of file, method, and version in which the method should have been deprecated.
+    """
+    decorated_functions = find_decorated_functions_in_codebase(codebase_path, decorator_name)
+
+    should_be_removed = []
+    for filepath, funcs in decorated_functions.items():
+        for func in funcs:
+            func_name = func['name']
+            args, kwargs = func['args'], func['kwargs']
+            if 'removal_version' in kwargs:
+                rem_ver = kwargs['removal_version']
+            else:
+                rem_ver = args[-1]
+            rem_ver = extract_version_from_tuple(rem_ver)
+            if rem_ver <= medcat_version:
+                should_be_removed.append((filepath, func_name, rem_ver))
+    return should_be_removed
+
+
+def _ver2str(ver: Tuple[int, int, int]) -> str:
+    maj, min, patch = ver
+    return f"v{maj}.{min}.{patch}"
+
+
+def main(args: List[str] = sys_argv[1:],
+         deprecated_decorator: Callable[[], Callable] = deprecated):
+    decorator_name = deprecated_decorator.__name__
+    pos_args = [arg for arg in args if not arg.startswith("-")]
+    codebase_path = 'medcat' if len(pos_args) <= 1 else pos_args[1]
+    print("arg0", repr(args[0]))
+    remove_ver_prefix = '--remove-prefix' in args
+    pure_ver = pos_args[0]
+    if remove_ver_prefix:
+        # remove v from (e.g) v1.12.0
+        pure_ver = pure_ver[1:]
+    medcat_version = tuple(int(s) for s in pure_ver.split("."))
+    compare_next_minor_release = '--next-version' in args
+
+    # pad out medcat varesions
+    # NOTE: Mostly so that e.g (1, 12, 0) <= (1, 12, 0) would be True.
+    #       Otherwise (1, 12, 0) <= (1, 12) would equate to False.
+    if len(medcat_version) < 3:
+        medcat_version = tuple(list(medcat_version) + [0,] * (3 - len(medcat_version)))
+    # NOTE: In main GHA workflow we know the current minor release
+    #       but after that release has been done, we (generally, but not always!)
+    #       want to start removing deprecated methods due to be removed before
+    #       the next minor release.
+    if compare_next_minor_release:
```

Comment:
for the next major version this is going to get pretty confused. No need to change this here, Just update the release doc on what to do perhaps?